### PR TITLE
Implement phaseAngleClock extension for 2 and 3 windings transformers

### PR DESCRIPTION
**Pull request by mathbagu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#18 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
The phaseAngleClock extension doesn't exist

**What is the new behavior (if this is a feature change)?**
The phaseAngleClock extension is available for both 2 and 3 windings transformer.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:
This PR doesn't fix the second part of the issue #18 that I will implement in another PR. The second part is not directly bounded to this evolution and is quiet hard to port in C++ due to the usage of raw types in Java (TapChanger). I don't know yet how to implement it.

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: extensions/iidm/include/powsybl/iidm/extensions/iidm/ThreeWindingsTransformerPhaseAngleClock.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+namespace iidm {
+
+class ThreeWindingsTransformerPhaseAngleClock : public Extension {
+public:  // Extension
+    const std::string& getName() const override;
+
+    const std::type_index& getType() const override;
+
+public:
+    ThreeWindingsTransformerPhaseAngleClock(ThreeWindingsTransformer& battery, unsigned long phaseAngleClockLeg2, unsigned long phaseAngleClockLeg3);
```
Comment (gautierbureau) [inline extensions/iidm/include/powsybl/iidm/extensions/iidm/ThreeWindingsTransformerPhaseAngleClock.hpp]: battery ? copy/paste wrong replace ?

File: extensions/iidm/include/powsybl/iidm/extensions/iidm/TwoWindingsTransformerPhaseAngleClock.hpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+namespace iidm {
+
+class TwoWindingsTransformerPhaseAngleClock : public Extension {
+public:  // Extension
+    const std::string& getName() const override;
+
+    const std::type_index& getType() const override;
+
+public:
+    TwoWindingsTransformerPhaseAngleClock(TwoWindingsTransformer& battery, unsigned long phaseAngleClock);
```
Comment (gautierbureau) [inline extensions/iidm/include/powsybl/iidm/extensions/iidm/TwoWindingsTransformerPhaseAngleClock.hpp]: idem

File: src/network/ThreeWindingsTransformerNetworkFactory.cpp [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        .setP0(0.0)
+        .setQ0(-10.6)
+        .add();
+    load11.getTerminal()
+        .setP(0.0)
+        .setQ(-10.6);
+
+    iidm::ThreeWindingsTransformer& twt = substation.newThreeWindingsTransformer()
+        .setId("3WT")
+        // TODO(mathbagu) .setRatedU0(132.0)
```
Comment (gautierbureau) [inline src/network/ThreeWindingsTransformerNetworkFactory.cpp]: Why TODO ?

Comment (mathbagu) [reply src/network/ThreeWindingsTransformerNetworkFactory.cpp]: This method doesn't exist yet. It appears in the new 3WT API (#48). This is a good catch. I have to think to fix it when this PR will be merged

File: extensions/iidm/include/powsybl/iidm/extensions/iidm/TwoWindingsTransformerPhaseAngleClock.hpp (hunk 2 above)
Comment (mathbagu) [reply extensions/iidm/include/powsybl/iidm/extensions/iidm/TwoWindingsTransformerPhaseAngleClock.hpp]: Good catch! I will fix it

File: extensions/iidm/include/powsybl/iidm/extensions/iidm/ThreeWindingsTransformerPhaseAngleClock.hpp (hunk 1 above)
Comment (mathbagu) [reply extensions/iidm/include/powsybl/iidm/extensions/iidm/ThreeWindingsTransformerPhaseAngleClock.hpp]: Good catch. I will fix it

**Pull request merged**